underpants.go

```diff
@@ -267,6 +267,25 @@ func urlFor(scheme, host string, r *http.Request) *url.URL {
   return &u
 }
 
+// Rewrite `Location` headers to point at the external hostname of a service if
+// the downstream uses its internal name.
+func rewriteRedirects(headers *http.Header, external string, internal string) {
+  location := headers.Get("Location")
+  if location == "" {
+    return
+  }
+
+  location_url, err := url.Parse(location)
+  if err != nil {
+    return
+  }
+
+  if location_url.Host == internal {
+    location_url.Host = external
```

Comment:
Would probably be good to do, but in our case with the "secure headers" turned on, HSTS takes care of it.  I'll add that in a bit.